crates/oxc_ast/src/ast/ts.rs

```diff
@@ -858,6 +859,7 @@ pub enum TSAccessibility {
 pub struct TSClassImplements<'a> {
     pub span: Span,
     pub expression: TSTypeName<'a>,
+    #[estree(rename = "typeArguments")]
```

Comment:
We should change the Rust field names directly for these renames.

Reply:
Nice, I was about to suggest that too :+1:
Not sure I've covered everything yet. I'll look up what typsciprt-eslint did in https://github.com/typescript-eslint/typescript-eslint/pull/5384